### ISSUE TITLE
ban `@final` on `Protocol`s

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#621]
it doesn't do anything so it's very likely a mistake

Code sample in [basedpyright playground](https://basedpyright.com/?pythonVersion=3.13&typeCheckingMode=all&reportUninitializedInstanceVariable=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYA0UwqAhhQFBMACdKjTpF9AznygAxMGAAURMCXIUAlAC4mUZVHrzMKGC268BUAEL0QilavWotpgEbnNLYKPUiwUALwGjY2UA)

```python
from typing import Protocol, final

@final
class Foo(Protocol):
    a: int


class Bar:
    a: int
    b: int

foo: Foo = Bar() # no error
```

**Comment by KotlinIsland** [comment]
i wouldn't say that it doesn't do anything, it does prevent explicit subtyping:
```py
from typing import Protocol, final

@final
class Foo(Protocol):
    a: int


class Bar(Foo, Protocol):  # Base class "Foo" is marked final and cannot be subclassed
    b: int
```
but yeah, i agree that it's very misleading

**Comment by KotlinIsland** [comment]
do we want `closed` protocols?

**Comment by DetachHead** [comment]
i dont think so, would it not be the exact same thing but with a different name?
